pkg/devicemapper/devmapper.go

```diff
@@ -342,7 +341,7 @@ func RemoveDevice(name string) error {
 		return fmt.Errorf("devicemapper: Error running RemoveDevice %s", err)
 	}
 
-	return nil
+	return UdevWait(cookie)
```

Comment:
I understand that docker developers like to use defer, but I assert that theres no need for it here.  If the goal is to capture any error from UdevWait and return it to the calling function, then I'm not sure why its worth blocking on a deferred function, when its more direct to just wait for the completion inline.